addons/crm_livechat/models/discuss_channel.py

```diff
@@ -72,11 +73,18 @@ def _convert_visitor_to_lead(self, partner, key):
         })
 
     def _get_livechat_session_fields_to_store(self):
+        def can_read_opportunity_ids(partner):
+            try:
+                partner.fetch(["opportunity_ids"])
+            except AccessError:
+                return False
+            else:
+                return True
```

Comment:
Does the access actually depend on each partner individually?

Can't we just check once globally if the model can be accessed, and append the field if it's ok?